src/enums/ClaimConditionType.ts

```diff
@@ -0,0 +1,27 @@
+enum ClaimConditionType {
+  StartTime,
+  LimitPerWallet,
+  MerkleRoot,
+  WaitTimeBetweenClaims,
+}
+
+/**
+ * Questions:
```

Comment:
@jakeloo @nkrishang need your input on all these questions, please. I'm trying to craft a `ClaimConditionsFactory`, but I need to understand the fundamental rules around all of our supported claim conditions first.